game/src/contrib/systems/XPSystem.java

```diff
@@ -2,18 +2,21 @@
 
 import contrib.components.XPComponent;
 
-import core.Game;
 import core.System;
 
 public class XPSystem extends System {
 
+    public XPSystem() {
+        super(XPComponent.class);
+    }
+
     @Override
-    public void update() {
-        Game.getEntities().stream()
-                .flatMap(e -> e.getComponent(XPComponent.class).stream())
+    public void execute() {
+        getEntityStream()
                 .forEach(
-                        component -> {
-                            XPComponent comp = (XPComponent) component;
+                        entity -> {
+                            XPComponent comp =
+                                    (XPComponent) entity.getComponent(XPComponent.class).get();
                             long xpLeft;
                             while ((xpLeft = comp.getXPToNextLevel()) <= 0) {
                                 this.performLevelUp(comp, (int) xpLeft);
```

Comment:
Wir haben einen Stream - und das ist so lange lesbar, wie wir in einem Stream einfach nur Methoden(referenzen) aufrufen. Wenn wir das wie hier mit prozeduralem Code mischen, wird es einfach nur unleserlich. Ich würde den Teil innerhalb des `forEach` in eine Hilfsmethode auslagern, die hier per Referenz aufgerufen wird.